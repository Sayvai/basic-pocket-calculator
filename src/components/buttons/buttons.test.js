import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Buttons from './buttons';

describe('Buttons component', () => {
  let props;

  beforeEach(() => {
    props = {
      onButtonPress: jest.fn(),
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render multiple buttons', () => {
    // when
    const { container } = render(<Buttons {...props} />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should call props.onButtonPress when a button is clicked', () => {
    // given
    render(<Buttons {...props} />);

    // when
    const buttonFive = screen.getByTestId('Five');
    fireEvent.click(buttonFive);

    // then
    expect(props.onButtonPress).toHaveBeenCalled(); // I would typically specifiy object detail, but React click events are pretty comprehensive..
  });
});