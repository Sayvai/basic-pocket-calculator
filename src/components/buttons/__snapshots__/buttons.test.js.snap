// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons component should render multiple buttons 1`] = `
<div>
  <div
    class="buttons"
    data-testid="component-buttons"
  >
    <button
      aria-label="Clear"
      class="button-clear"
      data-button-value="c"
      data-testid="Clear"
    >
      C
    </button>
    <button
      aria-label="Divide"
      data-button-operator="true"
      data-button-value="/"
      data-testid="Divide"
    >
      ÷
    </button>
    <button
      aria-label="Seven"
      data-button-value="7"
      data-testid="Seven"
    >
      7
    </button>
    <button
      aria-label="Eight"
      data-button-value="8"
      data-testid="Eight"
    >
      8
    </button>
    <button
      aria-label="Nine"
      data-button-value="9"
      data-testid="Nine"
    >
      9
    </button>
    <button
      aria-label="Multiply"
      data-button-operator="true"
      data-button-value="*"
      data-testid="Multiply"
    >
      ×
    </button>
    <button
      aria-label="Four"
      data-button-value="4"
      data-testid="Four"
    >
      4
    </button>
    <button
      aria-label="Five"
      data-button-value="5"
      data-testid="Five"
    >
      5
    </button>
    <button
      aria-label="Six"
      data-button-value="6"
      data-testid="Six"
    >
      6
    </button>
    <button
      aria-label="Subtract"
      data-button-operator="true"
      data-button-value="-"
      data-testid="Subtract"
    >
      −
    </button>
    <button
      aria-label="One"
      data-button-value="1"
      data-testid="One"
    >
      1
    </button>
    <button
      aria-label="Two"
      data-button-value="2"
      data-testid="Two"
    >
      2
    </button>
    <button
      aria-label="Three"
      data-button-value="3"
      data-testid="Three"
    >
      3
    </button>
    <button
      aria-label="Add"
      data-button-operator="true"
      data-button-value="+"
      data-testid="Add"
    >
      +
    </button>
    <button
      aria-label="Zero"
      class="button-zero"
      data-button-value="0"
      data-testid="Zero"
    >
      0
    </button>
    <button
      aria-label="Equal"
      class="button-equal"
      data-button-operator="true"
      data-button-value="="
      data-testid="Equal"
    >
      =
    </button>
  </div>
</div>
`;
